src/main/java/bio/terra/workspace/service/workspace/flight/DeleteWorkspaceAuthzStep.java

```diff
@@ -35,7 +34,11 @@ public StepResult doStep(FlightContext flightContext) throws RetryException {
   public StepResult undoStep(FlightContext flightContext) {
     // Sam does not allow Workspace ID re-use, so a delete really can't be undone. We retry on Sam
     // API errors in the do-step to try avoiding the undo step, but if we get this far there's
-    // nothing to do but tell Stairway we're stuck.
-    return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL);
+    // nothing to do but tell Stairway we're stuck and surface the error from the DO step.
+    FlightMap inputMap = flightContext.getInputParameters();
+    UUID workspaceID =
+        UUID.fromString(inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, String.class));
+    logger.error("Unable to undo deletion of workspace {} in WSM DB", workspaceID);
+    return flightContext.getResult();
```

Comment:
The previous pattern of returning `new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL);` was always a 500 because we returned FATAL with no exception. Stairway would ignore the result from the DO direction, and throw a 500 because it doesn't have an exception from the UNDO direction.

With this pattern, fatal flights will return whatever the error from the DO direction was (in this specific case, 400).